<launch>
  <machine name="local" address="localhost" default="true" />

  <param name="use_sim_time" value="true"/>
  <arg name="dataset" default="" />
  <arg name="rate" default="1.0" />

  <node name="vinySlam" pkg="slam_fmwk" type="viny_slam"
        cwd="node" required="true" output="screen">
    <remap from="laser_scan" to="/base_scan" />

    <!-- SLAM Components -->
    <param name="slam/occupancy_estimator/type" value="const" />

    <!-- Map -->
    <param name="slam/map/height_in_meters" value="100" />
    <param name="slam/map/width_in_meters" value="100" />
    <param name="slam/map/meters_per_cell" value="0.1" />

    <!-- Internal constants -->
    <param name="vinySlam/hole_width" value="0.3" /> <!-- w < 0 for dynamic -->
    <param name="ros/skip_exceeding_lsr_vals" value="false" />
    <param name="slam/occupancy_estimator/base_occupied/prob" value="0.95" />
    <param name="slam/occupancy_estimator/base_occupied/qual" value="0.04" />
    <param name="slam/occupancy_estimator/base_empty/prob" value="0.01" />
    <param name="slam/occupancy_estimator/base_empty/qual" value="0.003" />

    <!-- Monte Carlo Scan Matcher-->
    <!-- <param name="slam/scmtch/MC/seed" value="666666" /> -->
    <param name="slam/scmtch/MC/sigma_XY" value="0.2" />
    <param name="slam/scmtch/MC/sigma_theta" value="0.1" />
    <param name="slam/scmtch/MC/limit_of_bad_attempts" value="20" />
    <param name="slam/scmtch/MC/limit_of_total_attempts" value="100" />

    <!-- ROS constants -->
    <param name="ros/rviz/map_publishing_rate" value="5.0" />
    <param name="ros/tf/buffer_duration" value="5.0" /> <!-- in seconds-->
    <param name="ros/filter_queue_size" value="1000" />
    <param name="ros/subscribers_queue_size" value="1000" />
  </node>

  <node name="bag_adapter" pkg="slam_fmwk"
        type="wg_pr2_bag_adapter" required="true" output="screen">
    <param name="tf_map_frame_id" value="map" />
    <param name="tf_odom_frame_id" value="odom_combined" />

    <remap from="provided_grid_orig" to="/map_bag" />
    <remap from="provided_grid" to="/ground_truth_map" />

    <remap from="tf_in" to="/tf_bag" />
    <remap from="tf_out" to="/tf" />
  </node>

  <node name="player" pkg="rosbag" type="play"
        args="--clock --rate $(arg rate) $(arg path)" >
     <remap from="/map" to="/map_bag" />
     <remap from="/tf" to="/tf_bag" />
  </node>

  <node name="rviz" pkg="rviz" type="rviz" respawn="true"
        args="-d $(find slam_fmwk)/rviz/wg_pr2.rviz" />

</launch>
